### PR TITLE
Here is a stab at a terminations status

### PR DESCRIPTION
**Pull request by bhalonen**
I'm mostly a user of math opt, and don't have a terribly deep understanding. If it appears I am missing something obvious, I probably am.

### CODE REVIEW

File: src/MOI_wrapper.jl [hunk 1]
```diff
@@ -196,25 +196,58 @@ const _SUPPORTED_MODEL_ATTRIBUTES = Union{
...
     # MOI.Silent, # TODO
-    # MOI.TerminationStatus,
+    MOI.TerminationStatus,
     # MOI.PrimalStatus,
     # MOI.DualStatus
 }
-
+@enum HiGHS_ModelStatus begin
+    NOTSET 
+    LOADERROR
```
Comment (odow) [inline src/MOI_wrapper.jl:205]: This should be `LOAD_ERROR`.

File: src/MOI_wrapper.jl [hunk 2]
```diff
@@ -196,25 +196,58 @@ const _SUPPORTED_MODEL_ATTRIBUTES = Union{
     # MOI.RawStatusString,  # TODO
     MOI.ResultCount,
     # MOI.Silent, # TODO
-    # MOI.TerminationStatus,
+    MOI.TerminationStatus,
     # MOI.PrimalStatus,
     # MOI.DualStatus
 }
-
+@enum HiGHS_ModelStatus begin
```
Comment (odow) [inline src/MOI_wrapper.jl:203]: I would call this `HighsModelStatus` to match the C code: https://github.com/ERGO-Code/HiGHS/blob/bf336493c1353e6a60aca47eff353c76599a077b/src/lp_data/HConst.h#L114-L135

There's a question on whether we want to use https://github.com/JuliaInterop/CEnum.jl, but this is probably okay for now.

File: src/MOI_wrapper.jl [hunk 3]
```diff
@@ -196,25 +196,58 @@ const _SUPPORTED_MODEL_ATTRIBUTES = Union{
...
+    NOTSET
```
Comment (odow) [inline src/MOI_wrapper.jl:204]: Maybe for clarity, have `NOTSET = 0` to indicate the enum starts at 0 (even though this is the default).

**Review by odow — COMMENTED** [review]
A few minor things, but this is on the right track.

We also need `PrimalStatus` and `DualStatus`, which are probably just checking `HighsModelStatus` and returning `FEASIBLE_POINT` if `OPTIMAL`.

File: src/MOI_wrapper.jl [hunk 4]
```diff
@@ -196,25 +196,58 @@ const _SUPPORTED_MODEL_ATTRIBUTES = Union{
...
+function MOI.get(o::Optimizer, ::MOI.TerminationStatus)
+    status = HiGHS_ModelStatus(Highs_getModelStatus(o.inner, Cint(0)))
+    if status == OPTIMAL
+        return MOI.OPTIMAL
+    elseif status == PRIMAL_INFEASIBLE
+        return MOI.INFEASIBLE
+    elseif status == DUAL_INFEASIBLE
+        return MOI.DUAL_INFEASIBLE
+    elseif status == REACHED_ITERATION_LIMIT
+        return MOI.REACHED_ITERATION_LIMIT
```
Comment (blegat) [inline src/MOI_wrapper.jl:242]: `MOI.ITERATION_LIMIT`

**Review by odow — APPROVED** [review]
This should do for now. There are few questions, like what is the difference between `DUAL_INFEASIBLE` and `PRIMAL_UNBOUNDED` (the solver returns a feasible primal solution in the second case, but has no primal solution in the first?), but I have plans to put in some effort tidying up the wrapper over the next couple of weeks.

**Pull request merged**